pages/exercises/[lessonSlug].tsx

```diff
@@ -104,55 +108,71 @@ const Exercises: React.FC<QueryDataProps<GetExercisesQuery>> = ({
 }
 
 type ExerciseData = {
+  id: number
   problem: string
   answer: string
   explanation: string
 }
 
 type ExerciseProps = {
-  exercise: ExerciseData
-  setExerciseIndex: React.Dispatch<React.SetStateAction<number>>
   lessonTitle: string
-  hasPrevious: boolean
-  hasNext: boolean
-  submitUserAnswer: (userAnswer: string) => void
+  exercises: ExerciseData[]
+  userAnswers: Record<number, string>
+  submitUserAnswer: (exerciseId: number, userAnswer: string) => void
+  onExit: (userAnswers: Record<number, string>) => void
 }
 
 const Exercise = ({
-  exercise,
-  setExerciseIndex,
   lessonTitle,
-  hasPrevious,
-  hasNext,
-  submitUserAnswer
+  exercises,
+  userAnswers,
+  submitUserAnswer,
+  onExit
 }: ExerciseProps) => {
   const [answerShown, setAnswerShown] = useState(false)
   const [message, setMessage] = useState(Message.EMPTY)
+  const [exerciseIndex, setExerciseIndex] = useState(0)
+  const [localUserAnswers, setLocalUserAnswers] = useState(userAnswers)
+  const exercise = exercises[exerciseIndex]
+
+  const hasPrevious = exerciseIndex > 0
+  const hasNext = exerciseIndex < exercises.length - 1
 
   return (
     <div className={`mx-auto ${styles.exercise__container}`}>
       <button
         className="btn ps-0 d-flex align-items-center"
-        onClick={() => setExerciseIndex(-1)}
+        onClick={() => onExit(localUserAnswers)}
```

Comment:
We pass all the `localUserAnswers` so that the parent component can update `userAnswers` to reflect the current state.